querydsl-mongodb/src/main/java/com/querydsl/mongodb/document/MongodbDocumentSerializer.java

```diff
@@ -107,18 +111,30 @@ public Object visit(Operation<?> expr, Void context) {
             return asDocument(asDBKey(expr, 0), "");
 
         } else if (op == Ops.AND) {
-            Map<Object, Object> lhs = (Map<Object, Object>) handle(expr.getArg(0));
-            Map<Object, Object> rhs = (Map<Object, Object>) handle(expr.getArg(1));
-            if (Sets.intersection(lhs.keySet(), rhs.keySet()).isEmpty()) {
-                lhs.putAll(rhs);
-                return lhs;
-            } else {
-                List<Object> list = new ArrayList<Object>(2);
-                list.add(handle(expr.getArg(0)));
-                list.add(handle(expr.getArg(1)));
-                return asDocument("$and", list);
+            Queue<Map<Object, Object>> pendingDocuments = collectConnectorArgs("$and", expr);
```

Comment:
@mp911de do you know by any chance if this same fix ("Fix $and and $or with more than two arguments in final query document") is required for / applicable to the `MongodbSerializer`?

I am asking because of this issue report: https://github.com/querydsl/querydsl/issues/2638#issue-681738472

Reply:
Yes, that's exactly the fix for the problem. We faced the same issue in Spring Data MongoDB and fixed it there. See https://jira.spring.io/browse/DATAMONGO-2475 for reference.

Reply:
Cool, it's unrelated to the MongoDB document API, but I might consider porting some fixes from https://github.com/spring-projects/spring-data-mongodb/blob/master/spring-data-mongodb/src/main/java/org/springframework/data/mongodb/repository/support/MongodbDocumentSerializer.java back to here if you don't mind.

I see you also need the `NoMatch` operator there. Are there any other candidates for a backport that you're aware of?

Reply:
NoMatch is a marker that we've introduced to guard empty joins. Not really required for backporting.